cfme/configure/settings.py

```diff
@@ -23,109 +25,173 @@ class TimeProfileAddForm(View):
-    def __init__(self, description=None, scope=None, days=None, hours=None, timezone=None,
-                 appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.description = description
-        self.scope = scope
-        self.days = days
-        self.hours = hours
-        self.timezone = timezone
+class TimeProfileEditView(TimeProfileAddFormView):
+    save_edit_button = Button('Save')
 
-    def create(self, cancel=False):
-        view = navigate_to(self, 'All')
-        view.timeprofile_form.configuration.item_select('Add a new Time Profile')
-        view.timeprofile_form.fill({
-            'description': self.description,
-            'scope': self.scope,
-            'days': self.days,
-            'hours': self.hours,
-            'timezone': self.timezone,
-        })
-        if not cancel:
-            view.timeprofile_form.save_button.click()
-            view.flash.assert_message('Time Profile "{}" was saved'.format(self.description))
+
+@attr.s
+class TimeProfile(Updateable, BaseEntity):
+
+    description = attr.ib(default=None)
+    scope = attr.ib(default=None)
+    days = attr.ib(default=True)
+    hours = attr.ib(default=True)
+    timezone = attr.ib(default=None)
 
     def update(self, updates):
-        view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
+        """
+        This method is used for updating the time_profile
+
+        Args:
+            updates: It the object of the time_profile that we need to update.
+        """
+
+        view = navigate_to(self, 'Edit')
+        rows = view.time_profile_form.table
         for row in rows:
             if row.description.text == self.description:
                 row[0].check()
-        view.timeprofile_form.configuration.item_select('Edit selected Time Profile')
-        changed = view.timeprofile_form.fill({
+        view.time_profile_form.configuration.item_select('Edit selected Time Profile')
+        changed = view.time_profile_form.fill({
             'description': updates.get('description'),
             'scope': updates.get('scope'),
             'days': updates.get('days'),
             'hours': updates.get('hours'),
             'timezone': updates.get('timezone'),
         })
         if changed:
-            view.timeprofile_form.save_edit_button.click()
+            view.save_edit_button.click()
             view.flash.assert_message(
                 'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
 
     def copy(self, name=None):
-        view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
+        """
+        This method performs the copy of the provided time profile object.
+        Args:
+            name (str) : It's the name of the new copied time_profile.
+
+        return: It returns the object of the copied time_profile.
+        """
+
+        view = navigate_to(self.parent, 'All')
```

Comment:
**Required**

There should be a 'Copy' destination registered against the TimeProfile entity class, instead of custom navigation within this method.